### DISCUSSION TITLE
Migrate to TypeScript and React Hooks?

### DISCUSSION THREAD

**Discussion by mashirozx** [mastodon/mastodon#18893] [General]
As an old React project, most of our components were written in JaaScript and class component, and they are a bit outdated today. I think maybe it's time to migrate into the technology of modern and mainstream?

**Comment by shleeable** [comment]
Somebody tried previously. I think it's worth speaking with the big G about. https://github.com/mastodon/mastodon/pull/16210

**Reply by mashirozx** [reply]
Said to see it was closed 🥲

**Reply by shleeable** [reply]
I'd recommend contacting @Gargron and inquiring for more information.

**Comment by mjankowski** [comment]
It's three years later, but both of these migrations are ongoing.
